library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -466,6 +468,18 @@ letMatch (Sexp.List [name, bindings, body]) cont
       pure $ Sexp.list [name, Sexp.unGroupBy2 form, bod]
 letMatch _ _ = error "malformed let-match"
 
+-- | @handler@ follows the exact same logic as @let-match@
+handler :: HasClosure m => Sexp.T -> (Sexp.T -> m Sexp.T) -> m Sexp.T
+handler (Sexp.List [name, bindings, body]) cont
+  | Just nameSymb <- eleToSymbol name =
+    local @"closure" (Closure.insertGeneric nameSymb) $ do
+      -- this just makes it consistent with the lambdaCase case
+      let grouped = Sexp.groupBy2 bindings
+      form <- mapF (`matchMany` cont) grouped
+      bod <- cont body
+      pure $ Sexp.list [name, Sexp.unGroupBy2 form, bod]
+handler _ _ = error "malformed handler"
+
```

Comment:
Might be best to use `Structure` module for new additions. The old code needs to be refactored into this, but would be best to use the better infra now that we have it, even in a module with the old way.

Reply:
Makes sense